src/plugins/dashboard/public/application/hooks/use_dashboard_app_state.ts

```diff
@@ -320,7 +305,7 @@ export const useDashboardAppState = ({
                 setDashboardAppState((s) => ({ ...s, hasUnsavedChanges }));
 
                 unsavedChanges.viewMode = current.viewMode; // always push view mode into session store.
-                dashboardSessionStorage.setState(savedDashboardId, unsavedChanges);
+                dashboardSessionStorage.setState(unsavedChanges, savedDashboardId);
```

Comment:
Why was the order of the args changed here?

Reply:
When I was defining `DashboardSessionStorageServiceType`, I defined `setState` as `(newState: Partial<DashboardState>, id?: string) => void;` and, since `id` was optional, I had to make it the second argument with this type definition.

However, I modified this in 9631adb454336abb0ac4494bd4ea7b77c8fcf5e8 so that `setState` has the type `(id: string | undefined, newState: Partial<DashboardState>) => void;` instead and swapped the arguments back around since it probably makes more sense for `id` to be first in this case. Good catch 👍